aleth-vm/main.cpp

```diff
@@ -239,29 +238,31 @@ int main(int argc, char** argv)
     if (vm.count("input"))
         data = fromHex(vm["input"].as<string>());
     if (vm.count("code"))
-        code = fromHex(vm["code"].as<string>());
+        code = vm["code"].as<string>();
+    if (vm.count("codefile"))
+        codeFile = vm["codefile"].as<string>();
 
     // Read code from input file.
-    if (!inputFile.empty())
+    if (!codeFile.empty())
     {
-        if (!code.empty())
-            cerr << "--code argument overwritten by input file " << inputFile << '\n';
+        if (vm.count("code"))
+        {
+            cerr << "Options --code and --codefile shouldn't be used at the same time" << '\n';
+            return AlethErrors::BadConfigOption;
+        }
 
-        if (inputFile == "-")
-            for (int i = cin.get(); i != -1; i = cin.get())
-                code.push_back(static_cast<byte>(i));
+        if (codeFile == "-")
+            std::getline(std::cin, code);
         else
-            code = contents(inputFile);
+            code = contentsString(codeFile);
 
         try  // Try decoding from hex.
         {
-            std::string strCode{reinterpret_cast<char const*>(code.data()), code.size()};
-            strCode.erase(strCode.find_last_not_of(" \t\n\r") + 1);  // Right trim.
-            code = fromHex(strCode, WhenError::Throw);
+            code.erase(code.find_last_not_of(" \t\n\r") + 1);  // Right trim.
```

Comment:
Currently `try`/`catch` around this is useless (`erase` and `code` never throw), it was here to handle errors trown by `fromHex`.

It's fine that you moved `fromHex` into a single place for both `--code` and `--codefile`, but there needs to be hanlding of exceptions around it.

I guess you would need some new variable like `bytes codeBytes` and you would assign result `fromHex` to it inside `try/catch`, then pass `codeBytes` to `setCode`.

Also I find it strange that we ignored decoding errors before, so I'd say inside `catch` we should better output error and exit.

Reply:
@gumb0 Changes pushed in the latest commit, please review if I understood your requirements perfectly. Thanks.